test/org/javarosa/core/util/test/NumericEncodingTest.java

```diff
@@ -19,16 +19,18 @@
 import junit.framework.Test;
 import junit.framework.TestCase;
 import junit.framework.TestSuite;
-
 import org.javarosa.core.util.externalizable.ExtWrapIntEncoding;
 import org.javarosa.core.util.externalizable.ExtWrapIntEncodingSmall;
 import org.javarosa.core.util.externalizable.ExtWrapIntEncodingUniform;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-public class NumericEncodingTest extends TestCase  {
+public class NumericEncodingTest extends TestCase {
+    private static final Logger logger = LoggerFactory.getLogger(NumericEncodingTest.class);
 
     public NumericEncodingTest(String name) {
         super(name);
-        System.out.println("Running " + this.getClass().getName() + " test: " + name + "...");
+        logger.info("Running {} test: {}...", this.getClass().getName(), name);
```

Comment:
I'm wondering whether System.out might be the right interface for a test... I'm not sure! Might be odd to get it in a log file?

Reply:
I don't know... I guess that what we should really do is to migrate tests to junit4 and use proper test method names for such feedback messages.

Do you want me to revert the changes on tests?

Reply:
If you think sending them to the logging makes sense, I'm fine with leaving it. Agreed migrating the tests to junit4 would be ideal. Can that be a quick, surface change or will it require going deeper into each test file?

Reply:
I think the migration to junit4 is pretty straightforward thanks to the structure we have followed so far.

One would have to go through the all the 22 test classes and:
- Remove the extension on `TestCase`
- Add a `@Before` annotation on `setUp` and remove the call on `super.setUp()`
- Add a `@Test` annotation on all methods listed on `testMaster()`.
- Remove `testMaster()` and `suite()`
- Remove the class constructor

Reply:
Sounds great. @ggalmazor you want to just do it (in a PR after this one)?

Reply:
OK!

Reply:
So, I have to disagree with @lognaturel's original comment here. Using the logging framework in the tests is the better option IMO. The reason is that it gives you the flexibility to redirect output to a file by default (so you don't blast the console during automated testing), but enable it conditionally while running them during a test debugging session. Really, this is what these frameworks were invented for. Remember those commented `System.out`s you removed? Well, if you log them you never have to leave commented print statements... and... you can enable them at runtime without ever having to recompile the code.